music_assistant/server/providers/tidal/__init__.py

```diff
@@ -0,0 +1,382 @@
+        similar_tracks_obj = await get_similar_tracks(self._tidal_session, prov_track_id, limit)
+        tracks = []
+        for track_obj in similar_tracks_obj:
+            if track_obj.available:
+                track = parse_track(tidal_provider=self, track_obj=track_obj)
+                tracks.append(track)
+        return tracks
+
+    async def library_add(self, prov_item_id, media_type: MediaType):
+        """Add item to library."""
+        return await library_items_add_remove(
+            self._tidal_session,
+            self._tidal_user_id,
+            prov_item_id,
+            media_type,
+            add=True,
+        )
+
+    async def library_remove(self, prov_item_id, media_type: MediaType):
+        """Remove item from library."""
+        return await library_items_add_remove(
+            self._tidal_session,
+            self._tidal_user_id,
+            prov_item_id,
+            media_type,
+            add=False,
+        )
+
+    async def add_playlist_tracks(self, prov_playlist_id: str, prov_track_ids: list[str]):
+        """Add track(s) to playlist."""
+        return await add_remove_playlist_tracks(
+            self._tidal_session, prov_playlist_id, prov_track_ids, add=True
+        )
+
+    async def remove_playlist_tracks(
+        self, prov_playlist_id: str, positions_to_remove: tuple[int, ...]
+    ) -> None:
+        """Remove track(s) from playlist."""
+        prov_track_ids = []
+        async for track in self.get_playlist_tracks(prov_playlist_id):
+            if track.position in positions_to_remove:
+                prov_track_ids.append(track.item_id)
+            if len(prov_track_ids) == len(positions_to_remove):
+                break
+        return await add_remove_playlist_tracks(
+            self._tidal_session, prov_playlist_id, prov_track_ids, add=False
+        )
+
+    async def create_playlist(self, name: str) -> Playlist:  # type: ignore[return]
+        """Create a new playlist on provider with given name."""
+        playlist_obj = await create_playlist(self._tidal_session, self._tidal_user_id, name)
+        playlist = parse_playlist(tidal_provider=self, playlist_obj=playlist_obj)
+        return await self.mass.music.playlists.add_db_item(playlist)
+
+    async def get_stream_details(self, item_id: str) -> StreamDetails:
+        """Return the content details for the given track when it will be streamed."""
+        # make sure a valid track is requested.
+        track = await get_track(self._tidal_session, item_id)
+        url = await get_track_url(self._tidal_session, item_id)
+        if not track:
+            raise MediaNotFoundError(f"track {item_id} not found")
+        # make sure that the token is still valid by just requesting it
+        await validate_token_and_refresh(self._tidal_session, self.mass, self.config)
+        return StreamDetails(
+            item_id=track.id,
+            provider=self.instance_id,
+            content_type=ContentType.FLAC,
```

Comment:
for a follow up PR: make sure that the sample rate and bit depth is correct here

Reply:
Will do